.github/workflows/ci-pipeline.yml

```
@@ -8,4 +8,15 @@ jobs:
     steps:
     - uses: actions/checkout@v4
     - run: docker build --target test --tag todo_app .
-    - run: docker run todo_app
\ No newline at end of file
+    - run: docker run todo_app
+  deploy:
+    if: github.ref == 'refs/heads/main'
+    name: Deploy
+    needs: build
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v4
+    - run: docker login -u elliotbarnes -p ${{ secrets.DOCKER_TOKEN }}
```

You may notice the pipeline generates a couple of warnings here, the first of those can be easily resolved by switching the command slightly to use the `password-stdin` argument :
```suggestion
    - run: echo ${{ secrets.DOCKER_TOKEN }} | docker login -u elliotbarnes --password-stdin
```

But note that I don't think this is a particularly meaningful change when we're trusting the pipeline runners to be properly disposed of, and in fact as it stands the token is still injected raw into the command! To get value, we'd like want to convert the secret -> environment variable first, and then echo the env reference, something like:
```yml
    - run: echo $DOCKER_TOKEN | docker login -u elliotbarnes --password-stdin
      env: 
        DOCKER_TOKEN: ${{ secrets.DOCKER_TOKEN }}
```